oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/MeasuringDocumentStore.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jackrabbit.oak.plugins.document;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.jackrabbit.oak.cache.CacheStats;
+import org.apache.jackrabbit.oak.plugins.document.cache.CacheInvalidationStats;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * DocumentStore wrapper which can measure time spent in any call.
+ */
+public class MeasuringDocumentStore implements DocumentStore {
```

Comment:
There is already TimingDocumentStoreWrapper with a very similar feature. Would it make sense to use the existing class and adjust if needed?

Reply:
Yes, I did see that one but somehow as it was more complex and production code I considered it less intrusive to add a dedicated, simpler test code one. 

But revisiting it, I suppose what's missing is really only `resetMeasurement` and `getAndResetMeasurement` - so that could be added to TimingDocumentStoreWrapper indeed ..

Reply:
Added in [this commit](https://github.com/apache/jackrabbit-oak/pull/260/commits/a88af85d12fb4a56afb98771f2b3f986428ba3d3)